common/src/main/java/gov/cms/ab2d/common/service/JobServiceImpl.java

```diff
@@ -48,6 +59,28 @@ public Job createJob(String resourceTypes, String url) {
         job.setProgress(0);
         job.setUser(userService.getCurrentUser());
 
+        if (contractNumber != null) {
+            contractRepository.findContractByContractNumber(contractNumber).ifPresentOrElse(contractFound -> {
+                User user = userService.getCurrentUser();
+                Sponsor userSponsor = user.getSponsor();
+                Sponsor contractSponsor = contractFound.getSponsor();
+                if (!userSponsor.getHpmsId().equals(contractSponsor.getHpmsId())) {
+                    log.error("User {} does not have permissions for contract {}", user.getUsername(), contractFound.getContractNumber());
+                    throw new UserPermissionsException("User " + user.getUsername() + " does not have permissions for contract " + contractFound.getContractNumber());
+                }
+
+                if (contractFound.getAttestedOn() == null) {
+                    log.error("Contract {} has not been attested for", contractFound.getContractNumber());
+                    throw new ContractHasNotBeenAttestedException("Contract " + contractFound.getContractNumber() + " has not been attested for");
+                }
+
+                job.setContract(contractFound);
+            }, () -> {
+                log.error("Contract {} was not found", contractNumber);
+                throw new ResourceNotFoundException("Contract " + contractNumber + " was not found");
+            });
+        }
+
```

Comment:
You can greatly simplify this if you use `userSponsor.getAggregatedAttestedContracts()`.  This will give all of the attested contracts for the Sponsor taking care of parent/child stuff. 

Of course you will not be able to tell whether the contract number if wrong, or whether it exists but is unattested. I think that's fine and actually makes our error handling simpler. Just throw a general `No attested contract with contract number NNNN found for the user`.